### PR TITLE
Mount the entire user SSH directory into build container

### PR DESCRIPTION
**Pull request by jacksgt**
This enables the user to use an key file format (RSA, ED25519, ...).
Additionally, it allows more complex workflows (such as different SSH keys for
specfic sites, such as Github or Bitbucket), since the .ssh/config
file is also mounted into the container.

Fixes https://github.com/UnitedIncome/serverless-python-requirements/issues/488

### CODE REVIEW

File: lib/pip.js [hunk 1]
```diff
@@ -201,10 +201,7 @@ function installRequirements(targetFolder, serverless, options) {
     if (options.dockerSsh) {
       // Mount necessary ssh files to work with private repos
       dockerCmd.push(
-        '-v',
```
Comment (miketheman) [inline lib/pip.js]: Will this work if all the -v directives are removed? I think you need at least one.

Comment (jacksgt) [reply lib/pip.js]: Oh, I'm sorry, I copied wrong from my local installation. Thanks!

**Comment by jacksgt** [comment]
Adding some tests sounds like a good idea, however I'm not very experienced with JS tests.
Basically what I would do is the following:
1. create a file in ~/.ssh/id_foobar
2. run sls with dockerizePip option
3. check if the file is mounted inside the docker container

**Comment by jacksgt** [comment]
@miketheman Hey, I tried adding a test, but I'm not sure why the test does not produce a ZIP file (and subsequently fails).
Could you look into it?

**Comment by jacksgt** [comment]
@miketheman After a few more attempts I got it working. Could you review the PR again when you find the time? Thanks.

**Review by miketheman — DISMISSED** [review]
This looks great, thanks for taking the time to write the test!
I know that this branch is out of date from `master` and @bsamuel-ui has been focusing on changing the CI/CD from CircleCI to GitHub Actions - so I'd recommend rebasing/merging after that PR is on `master` so as to exercise this one more time in the new CI before merging.

**Comment by jacksgt** [comment]
Since there hasn't been any activity on the github actions PR for a two weeks now, could we merge this one in anyway?

**Comment by bsamuel-ui** [comment]
@jacksgt I was looking at that merge conflict, I could sort of resolve that through the editor, but I'm just as likely to screw it up. You just need to make sure your new test is present and is async, and as soon as you can do that we can merge this.

**Comment by jacksgt** [comment]
@bsamuel-ui Hey, thanks for getting back on this one and keeping the project going!

I rebased on master, but now the test is failing with a git error code that seems to indicate there is no public/private ssh key present on the host machine.
Since the testcase worked on the old CI, I assume that something changed in the setup.
Any idea how to fix this?

**Comment by bsamuel-ui** [comment]
Seeing a bunch of issues here that aren't in other branches. I'm going to take some time to dig into it on Tuesday per #550

File: test.js [hunk 2]
```diff
@@ -579,6 +581,35 @@ test(
...
+      '--dockerSsh=true',
+      '--fileName=requirements-w-git-ssh.txt',
+      'package'
+    ]);
+    const zipfiles = listZipFiles('.serverless/sls-py-req-test.zip');
+    // check if the requirement is actually in the archive
+    t.true(zipfiles.includes(`boto3/__init__.py`), 'boto3 is packaged via ssh');
+    t.end();
+  },
+  { skip: !canUseDocker() }
```
Comment (bsamuel-ui) [inline test.js:610]: Have you tested this on Windows?

Comment (jacksgt) [reply test.js:610]: Nope, not sure if its gonna work on Windows.

Comment (bsamuel-ui) [reply test.js:610]: Then the skip section should be skipping the test on Windows, and you can create an issue noting that this functionality doesn't work on Windows yet.

**Comment by tonythree** [comment]
Hey, having this issue as well.
Is this going to be merged at some point?
also, is there any workaround for this?

**Comment by martinezpl** [comment]
Any move with this? This is the current source code:

```javascript
// Mount necessary ssh files to work with private repos
      if (options.dockerSsh) {
        // Mount necessary ssh files to work with private repos
        dockerCmd.push(
          '-v',
          `${process.env.HOME}/.ssh/id_rsa:/root/.ssh/id_rsa:z`,
          '-v',
          `${process.env.HOME}/.ssh/known_hosts:/root/.ssh/known_hosts:z`,
          '-v',
          `${process.env.SSH_AUTH_SOCK}:/tmp/ssh_sock:z`,
          '-e',
          'SSH_AUTH_SOCK=/tmp/ssh_sock'
        );
      }
```

RSA keys are considered legacy, I don't think it's reasonable to continue expecting (or forcing, pretty much) this format.

**Comment by pgrzesik** [comment]
Hey @martinezpl - this PR has been dead for a long time - if you'd like to take over and propose a new one, feel free to do so 💯

**Pull request closed without merging**